### PR TITLE
[AIRFLOW-1763] Fix S3TaskHandler unit tests

### PR DESCRIPTION
**Pull request by andyxhadji**
Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1763

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
- Fixes & runs breaking tests for s3 task handler

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
- s3taskhandler tests in test_logging.py

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: tests/utils/log/test_logging.py [hunk 1]
```diff
@@ -21,72 +21,84 @@
...
         self.hook_mock.assert_called_once_with('')
 
     def test_init_raises(self):
         self.hook_mock.side_effect = Exception('Failed to connect')
-        handler = S3TaskHandler()
+        handler = self.s3_task_handler
         with mock.patch.object(handler.log, 'error') as mock_error:
             # Initialize the hook
-            handler.hook()
+            handler.hook
```
Comment (ashb) [inline tests/utils/log/test_logging.py]: Is this change correct? It looks suspect.

File: tests/utils/log/test_logging.py [hunk 2]
```diff
@@ -21,72 +21,84 @@
...
+        self.filename_template = ''
         self.hook_patcher = mock.patch("airflow.hooks.S3_hook.S3Hook")
         self.hook_mock = self.hook_patcher.start()
         self.hook_inst_mock = self.hook_mock.return_value
         self.hook_key_mock = self.hook_inst_mock.get_key.return_value
         self.hook_key_mock.get_contents_as_string.return_value.decode.\
             return_value = 'content'
+        class UnitTestS3TaskHandler(S3TaskHandler):
+            def test_instance(self):
+                return True
```
Comment (ashb) [inline tests/utils/log/test_logging.py]: What is this subclass and method for? It doesn't appear to be used anywhere in the codebase.

Comment (ashb) [reply tests/utils/log/test_logging.py]: (the test_instance method isn't used that I can see, the class is just used in here.)

Comment (andyxhadji) [reply tests/utils/log/test_logging.py]: Ah yea - I copied from another test I was building but turns out I didn't need it, thanks for pointing it out

File: tests/utils/log/test_logging.py (hunk 1 above)
Comment (andyxhadji) [reply tests/utils/log/test_logging.py]: The hook method in the handler is now a property, hence the change here!

**Comment by bolkedebruin** [comment]
Can you rename the file to mirror the file it is testing?

**Comment by andyxhadji** [comment]
@bolkedebruin File renamed and updated branch!

**Pull request closed without merging**